lib/libbackscrub.cc

```diff
@@ -99,94 +122,125 @@ static normalization_t get_normalization(modeltype_t type) {
 }
 
 // deeplabv3 classes
+// TODO: read from model metadata file
 static const std::vector<std::string> labels = { "background", "aeroplane", "bicycle", "bird", "boat", "bottle", "bus", "car", "cat", "chair", "cow", "dining table", "dog", "horse", "motorbike", "person", "potted plant", "sheep", "sofa", "train", "tv" };
 // label number of "person" for DeepLab v3+ model
 static const size_t cnum = labels.size();
 static const size_t pers = std::distance(labels.begin(), std::find(labels.begin(),labels.end(),"person"));
 
-int init_tensorflow(calcinfo_t &info) {
-	// Deallocate if not done already
-	if (info.backscrub_ctx)
-		drop_tensorflow(info);
+void *bs_maskgen_new(
+	// Required parameters
+	const char *modelname,
+	size_t threads,
+	size_t width,
+	size_t height,
+	// Optional (nullable) callbacks with caller-provided context
+	// ..debug output
+	void (*ondebug)(void *ctx, const char *fmt, va_list ap),
+	// ..after preparing video frame
+	void (*onprep)(void *ctx),
+	// ..after running inference
+	void (*oninfer)(void *ctx),
+	// ..after generating mask
+	void (*onmask)(void *ctx),
+	// ..the returned context
+	void *caller_ctx
+	) {
```

Comment:
Maybe skip the indentation for this line to make the end of the argument list visually more obvious.
```suggestion
) {
```

Reply:
Gah, VScode missed one :smile: will fix.